lib/block-supports/elements.php

```diff
@@ -0,0 +1,65 @@
+<?php
+/**
+ * Elements styles block support.
+ *
+ * @package gutenberg
+ */
+
+/**
+ * Render the elements stylesheet.
+ *
+ * @param  string $block_content Rendered block content.
+ * @param  array  $block         Block object.
+ * @return string                Filtered block content.
+ */
+function gutenberg_render_elements_support( $block_content, $block ) {
+	$block_type = WP_Block_Type_Registry::get_instance()->get_registered( $block['blockName'] );
+	$link_color = _wp_array_get( $block['attrs'], array( 'style', 'elements', 'link', 'color', 'text' ), null );
+
+	/*
+	* For now we only care about link color.
+	* This code in the future when we have a public API
+	* should take advantage of WP_Theme_JSON::compute_style_properties
+	* and work for any element and style.
+	*/
+	if ( null === $link_color ) {
+		return $block_content;
+	}
+
+	$class_name = 'wp-elements-' . uniqid();
+
+	if ( strpos( $link_color, 'var:preset|color|' ) !== false ) {
+		// Get the name from the string and add proper styles.
+		$index_to_splice = strrpos( $link_color, '|' ) + 1;
+		$link_color_name = substr( $link_color, $index_to_splice );
+		$link_color      = "var(--wp--preset--color--$link_color_name)";
+	}
+	$link_color_declaration = esc_html( safecss_filter_attr( "color: $link_color" ) );
+
+	$style = "<style>.$class_name a{" . $link_color_declaration . ";}</style>\n";
```

Comment:
```suggestion
	$style = "<style>.$class_name a{" . $link_color_declaration . " !important;}</style>\n";
```

This suggestion fixes this use case:

- TT1-blocks.
- Add a paragraph with text and link colors applied.

The expected result is that both colors are applied but the link is not. This is because TT1-blocks has this rule `p.has-text-color a { color: inherit; }` with higher specificity than the class we add. While this is a theme issue that should be updated once this PR lands, we also need to cover for this sort of issues with themes. The way we address this in preset classes is by adding `!important` to the declaration (user styles are to be used above theme styles).

Reply:
I think we should probably fix the issue in the theme, using important for link color causes inconsistencies e.g: why are we using important for link color but not for other styles?
I guess we can temporarily include the important but we should plan on fixing the core themes and removing the important.

Reply:
> I think we should probably fix the issue in the theme, using important for link color causes inconsistencies e.g: why are we using important for link color but not for other styles?

We actually do for other presets as well. See https://github.com/WordPress/gutenberg/pull/29533 for details and rationale.